test/rules/no-promise-without-done-fail.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const rule = require('../../lib/rules/no-promise-without-done-fail')
+const RuleTester = require('eslint').RuleTester
+const ruleTester = new RuleTester()
+
+ruleTester.run('no-promise-without-done-fail', rule, {
+  valid: [
+    {
+      code: 'it("", function(done) { somethingAsync.then(function (res) { expect(res).toBe(true); done();}).catch(done.fail)})'
```

Comment:
would the rule work with arrow functions 
`it('', (done) => {
})`

Reply:
Should work, I didn't limit the solution to any type of function

Reply:
if it does, please add tests cases with arrow function

Reply:
@DianaSuvorova added arrow function test cases